juno.mk

```diff
@@ -0,0 +1,192 @@
+DEBUG ?= 0
+
+-include common.mk
+
+################################################################################
+# Mandatory definition to use common.mk
+################################################################################
+CROSS_COMPILE_NS_USER	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+CROSS_COMPILE_NS_KERNEL	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+CROSS_COMPILE_S_USER	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+CROSS_COMPILE_S_KERNEL	?= "$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+OPTEE_OS_BIN		?= $(OPTEE_OS_PATH)/out/arm-plat-vexpress/core/tee.bin
+OPTEE_OS_TA_DEV_KIT_DIR	?= $(OPTEE_OS_PATH)/out/arm-plat-vexpress/export-ta_arm64
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+ARM_TF_PATH		?= $(ROOT)/arm-trusted-firmware
+
+U-BOOT_PATH		?= $(ROOT)/u-boot
+U-BOOT_BIN		?= $(U-BOOT_PATH)/u-boot.bin
+
+################################################################################
+# Targets
+################################################################################
+all: arm-tf u-boot linux optee-os optee-client optee-linuxdriver generate-dtb xtest update_rootfs
+all-clean: arm-tf-clean busybox-clean u-boot-clean optee-os-clean \
+	optee-client-clean optee-linuxdriver-clean
+
+
+-include toolchain.mk
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+ARM_TF_EXPORTS ?= \
+	CFLAGS="-O0 -gdwarf-2" \
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_NONE_CROSS_COMPILE)"
+
+ARM_TF_FLAGS ?= \
+	SCP_BL2=$(ROOT)/vexpress-firmware/SOFTWARE/bl30.bin \
+	BL32=$(OPTEE_OS_BIN) \
+	BL33=$(U-BOOT_BIN) \
+	DEBUG=0 \
+	ARM_TSP_RAM_LOCATION=dram \
+	PLAT=juno \
+	SPD=opteed
+
+arm-tf: optee-os u-boot
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip
+
+arm-tf-clean:
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) clean
+
+################################################################################
+# Busybox
+################################################################################
+BUSYBOX_COMMON_TARGET = fvp-aarch64
+BUSYBOX_CLEAN_COMMON_TARGET = fvp-aarch64 clean
+BUSYBOX_COMMON_CCDIR = $(AARCH64_PATH)
+
+busybox: busybox-common
+
+busybox-clean: busybox-clean-common
+
+busybox-cleaner: busybox-cleaner-common
+
+################################################################################
+# Das U-Boot
+################################################################################
+
+U-BOOT_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+U-BOOT_DEFCONFIG_FILES := \
+	$(U-BOOT_PATH)/configs/vexpress_aemv8a_juno_defconfig \
+	$(ROOT)/kconfigs/u-boot_juno.conf
```

Comment:
Is this line correct? Other local kconfig files (at least for kernel) are stored in build/kconfig/ ... which seems to be where you have put it when looking at the files further down.